rfcs/0005-artifact-revision-and-digest/README.md

```diff
@@ -0,0 +1,352 @@
+# RFC-0005 Artifact `Revision` format and introduction of `Digest`
+
+**Status:** provisional
+
+<!--
+Status represents the current state of the RFC.
+Must be one of `provisional`, `implementable`, `implemented`, `deferred`, `rejected`, `withdrawn`, or `replaced`.
+-->
+
+**Creation date:** 2022-10-20
+
+**Last update:** 2022-11-02
+
+## Summary
+
+This RFC proposes to establish a canonical `Revision` format for an `Artifact`
+which points to a specific revision represented as a checksum (e.g. an OCI
+manifest digest or Git commit SHA) of a named pointer (e.g. an OCI image tag or
+Git tag). In addition, it proposes to include the algorithm name (e.g.
+`sha256`) as a prefix to an advertised checksum for an `Artifact` and
+further referring to it as a `Digest`, deprecating the `Checksum` field.
+
+## Motivation
+
+The current `Artifact` type's `Revision` field format is not "officially"
+standardized (albeit assumed throughout our code bases), and has mostly been
+derived from `GitRepository` which uses `/` as a separator between the named
+pointer (a Git branch or tag) and a specific (SHA-1, or theoretical SHA-256)
+revision.
+
+Since the introduction of `OCIRepository` and with the recent changes around
+`HelmChart` objects to allow the consumption of charts from OCI registries,
+this could be seen as outdated or confusing due to the format differing from
+the canonical format used by OCI, which is `<tag>@<algo>:<checksum>` (the
```

Comment:
I am not sure I understand the meanings of `OCI tag` in the document.

The way to  address an `image manifest` in the different OCI specs is  using a `name` and a `reference`.  The `distribution spec` states:
>  \<name\> refers to the namespace of the repository. \<reference\> MUST be either (a) the digest of the manifest or (b) a tag.

We would then have either `<name>:<tag>` or `<name>@sha256:<digest>`.

My understanding is that we want to use the second form, `<name>@sha256:<digest>`.

Reply:
You are right, I mixed something up while writing this bit and skimming through the OCI specs. The correct format is `<name>@<digest>`, as a "digest" includes the algorithm the checksum was calculated with (see [here](https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests)). Which is why I used `@<algo>:<checksum>`.

I'll replace the tag reference(s) with "name".